src/api/archive_unstable_storageDiff.md

```diff
@@ -0,0 +1,114 @@
+# archive_unstable_storageDiff
+
+**Parameters**:
+
+- `currentHash`: String containing a hexadecimal-encoded hash of the header of the block whose storage difference will be retrieved. The storage difference is calculated between the `currentHash` block and the parent of the `currentHash` block.
+
+- `fromHash` (optional): String containing a hexadecimal-encoded hash of the header of the block from which the storage difference will be calculated. When this parameter is provided, the storage difference is calculated between the `currentHash` block and the `fromHash` block. If this parameter is not provided, the storage difference is calculated between the `currentHash` block and the parent of the `currentHash` block.
+
+- `items` (optional): JSON object containing the following fields:
+
+    ```json
+    "items": {
+        "prefixes": [
+            {
+                "key": "0x...",
+                "type": "value" | "hash" | "none",
+            },
+        ],
+        "excludeKeyPrefixes": [
+            "0x...",
+        ],
+        "childTrie": "0x...",
+    }
+    ```
+
+  If the `items` parameter is not provided, the storage difference is calculated for all keys of the main storage trie.
+  - `prefixes` (optional): Array of JSON objects describing how the storage difference will be calculated. Each object contains the following fields:
+    - `key`: String containing the hexadecimal-encoded key prefix under which the storage difference is calculated.
+    - `type`: String equal to one of: `value`, `hash`, `none`.
+  - containing the key prefixes for which the storage difference will be calculated. If this parameter is not provided, the storage difference is calculated for all keys.
+  - `excludeKeyPrefixes` (optional): Array of strings containing the key prefixes for which the storage difference will not be calculated. If this parameter is not provided, the storage difference is calculated for all keys.
+  - `childTrie` (optional): A string containing the hexadecimal-encoded key of the child trie of the "default" namespace. If this parameter is not provided, the storage difference is calculated for the main storage.
+
+**Return value**: String containing an opaque value representing the operation.
```

Comment:
For the sake of simplicity, I've been trying to avoid subscriptions for all `archive`-related functions.
The subscription functions are important for `chainHead`, because you want be able to cancel operations in order to keep bandwidth usage to a minimum, and because you want to be able to cancel operations related to blocks that end up not being canonical.
However, none of these two things apply here (and don't apply for any `archive`-prefix function), so I don't really see the point of having a subscription.

Reply:
I assumed that the point of the subscription here is that calculating the diffs can be quite slow, so by streaming them (there can be a bunch of prefixes asked for) the user can `stop` them (and be asked to `continue`) if they find what they need or decide they don't need the server to continue working?

I agree though that it'd be much nicer to avoid the subscriptions though, so if this isn't an issue then it'd be great to simplify it!

Reply:
> if they find what they need or decide they don't need the server to continue working?

In my opinion it is a better idea for the client to send queries progressively, for example query from block 1 to 100, then block 101 to 200, then block 201 to 300, etc.
It is not so different to do so compared to sending a "continue", except that the client has more control.

Reply:
I do like this approach! Turning this subscription into a method is easier to follow from the user's perspective. I'll do some performance measurements for storage differences once we implement this in Substrate.

We could probably add a disk cache to make queries a bit faster if that turns to be a performance issue